tools/Import/Importer.cs

```diff
@@ -0,0 +1,271 @@
+            {
+                var localSw = Stopwatch.StartNew();
+                var localResourceCount = 0L;
+                var blobRangeIndex = blobRangeInt.Item1;
+                var blobRange = blobRangeInt.Item2;
+                var localLogPrefix = $"Reader={reader}.BlobRange=[{NumberOfBlobsToSkip + (blobRangeIndex * BlobRangeSize) + 1}-{NumberOfBlobsToSkip + (blobRangeIndex * BlobRangeSize) + blobRange.Count}]";
+                Console.WriteLine($"{localLogPrefix}: Starting...");
+
+                BatchExtensions.ExecuteInParallelBatches(GetLinesInBlobRange(blobRange, localLogPrefix), WriteThreads / ReadThreads, 100, (thread, lineBatch) =>
+                {
+                    Interlocked.Increment(ref writers);
+                    foreach (var line in lineBatch.Item2)
+                    {
+                        Interlocked.Increment(ref globalResourceCount);
+                        Interlocked.Increment(ref localResourceCount);
+                        PutResource(line, ref localResourceCount, ref epCalls, ref waits);
+                        if (swReport.Elapsed.TotalSeconds > ReportingPeriodSec)
+                        {
+                            lock (swReport)
+                            {
+                                if (swReport.Elapsed.TotalSeconds > ReportingPeriodSec)
+                                {
+                                    var resWritten = Interlocked.Read(ref globalResourceCount);
+                                    var resRead = Interlocked.Read(ref resourcesRead);
+                                    Console.WriteLine($"{globalLogPrefix}.Readers={Interlocked.Read(ref readers)}.Writers={Interlocked.Read(ref writers)}.EndPointCalls={Interlocked.Read(ref epCalls)}.Waits={Interlocked.Read(ref waits)}: reads={resRead} writes={resWritten} secs={(int)swWrites.Elapsed.TotalSeconds} read-speed={(int)(resRead / swReads.Elapsed.TotalSeconds)} lines/sec write-speed={(int)(resWritten / swWrites.Elapsed.TotalSeconds)} res/sec");
+                                    swReport.Restart();
+                                }
+                            }
+                        }
+                    }
+
+                    Interlocked.Decrement(ref writers);
+                });
+                Console.WriteLine($"{localLogPrefix}: Completed writes. Total={localResourceCount} secs={(int)localSw.Elapsed.TotalSeconds} speed={(int)(localResourceCount / localSw.Elapsed.TotalSeconds)} res/sec");
+            });
+            Console.WriteLine($"{globalLogPrefix}.Readers={readers}.Writers={writers}.EndPointCalls={epCalls}.Waits={waits}: total reads={resourcesRead} total writes={globalResourceCount} secs={(int)swWrites.Elapsed.TotalSeconds} read-speed={(int)(resourcesRead / swReads.Elapsed.TotalSeconds)} lines/sec write-speed={(int)(globalResourceCount / swWrites.Elapsed.TotalSeconds)} res/sec");
```

Comment:
I'd provide a summary report by defining a string list and adding reads and writes messages to it.

            var doneMsg = $"{globalLogPrefix}.Readers={readers}.Writers={writers}.EndPointCalls={epCalls}.Waits={waits}: total reads={resourcesRead}; total writes={globalResourceCount}; total secs={(int)swWrites.Elapsed.TotalSeconds}; read-speed={(int)(resourcesRead / swReads.Elapsed.TotalSeconds)} lines/sec; write-speed={(int)(globalResourceCount / swWrites.Elapsed.TotalSeconds)} res/sec";
            summaryReport.Add(doneMsg);
            //Console.WriteLine(doneMsg);
            Console.WriteLine("\n*** Summary Report ***");
            Console.WriteLine($"{DateTime.Now.ToLongDateString()}");
            foreach (var msg in summaryReport) Console.WriteLine(msg);

Reply:
This summary is provided on reporting interval many times during run. I want it to be single line.